backend/app/db/supabase/users_service.py

```
@@ -0,0 +1,205 @@
+            "discord_id": discord_id,
+            "display_name": display_name,
+            "discord_username": discord_username,
+            "avatar_url": avatar_url,
+            "preferred_languages": [],
+            "created_at": datetime.now().isoformat(),
+            "updated_at": datetime.now().isoformat()
+        }
+        insert_res = await supabase.table("users").insert(new_user_data).execute()
+        if not insert_res.data:
+            raise Exception("Failed to create new user in database.")
+        return User(**insert_res.data[0])
```

Pylint (3.3.7)

[refactor] Unnecessary "else" after "return", remove the "else" and de-indent the code inside it

(R1705)